### ISSUE TITLE
Connection times out randomly on persistent connections when running more than 1 instance of traefik ingress controller.

### ISSUE THREAD

**Issue by shaggykp** [traefik/traefik#7672]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
Our Traefik ingress controller (v2.3.2) running in our K8s (v1.18) cluster is closing our persistent http/tcp connections.  The duration of time the connection remains open is inconsistent from one connection to the next.  We use MetalLB as our entry point in front of the Traefik ingress controller.  We seem to see this behavior when running more than 1 instance of the traefik ingress controller.

In our helm chart we have tried adding the following as addition arguments to the controller:

-  "--serverTransport.forwardingTimeouts.idleConnTimeout=0"
-  "--serverTransport.forwardingTimeouts.dialTimeout=0"
-  "--serverTransport.forwardTimeouts.idleConnTimeout=500"

Nothing seems to work.  

We were using the nginx ingress controller in our K8s cluster before transitioning to Traefik.  We never had any issues in this regard with the nginx ingress controller.

### What did you expect to see?
No connection timeouts regardless of the number of how many replicas of the traefik.  ingress controller are running.

### What did you see instead?
Connections randomly timeout when running more than 1 instance of the ingress controller.

### Output of `traefik version`: (_What version of Traefik are you using?_)
kubernetes 1.18
traefik 2.3.2

```
(paste your output here)
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
# (paste your configuration here)
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
(paste your output here)
```

**Comment by SantoDE** [comment]
Hello @shaggykp,

can you share a reproducible example with us please? :) We´d like to give it a run down and see if its something we can reproduce.

**Comment by shaggykp** [comment]
Please see attached files for my traefik deployment and service definitions.  

key points:
- traefik is running behind metallb
- there are 2 services (tcp and udp) both configured to use the same external IP
- externalTrafficPolicy is set to "Local" in both the service definitions
- the selector label for both services is the same.
- the test deployment is running with 5 instances of the traefik pods
- the deployment is using Traefik v 2.3.2, K8s v1.18
- there is a demo client application (external to the K8s cluster) that opens a persistent keepalive connection to a backend http service running inside the K8s cluster and sends an http request every 1 second.
- 
After some extensive testing, what I'm finding that when I define both the tcp and udp services, the http connection from the client app to the ingress is dropped but the it appears that the TCP connection between the traefik pod and the backend service is still alive.

If I delete the udp service and just keep the tcp traefik service configured to use the external IP, all seems to be happy and I see no dropped connections.
[ingress-deployment.txt](https://github.com/traefik/traefik/files/5732078/ingress-deployment.txt)
[ingress-udp-svc.txt](https://github.com/traefik/traefik/files/5732079/ingress-udp-svc.txt)
[ingress-tcp-svc.txt](https://github.com/traefik/traefik/files/5732080/ingress-tcp-svc.txt)

**Comment by shaggykp** [comment]
Any thoughts or updates yet on a possible solution?

**Comment by jspdown** [comment]
Hello @shaggykp. I've setup an environment but I'm not able to reproduce the issue. To go further on this I will need:
- Complete configuration (Traefik deployment, Ingress and services) 
- Complete logs in debug mode.

In the meantime, could you tried to incrementally simplify your setup so we can isolate better the problem?

**Comment by shaggykp** [comment]
Hi,

Thanks for looking into this.  The yaml files I included in my post is the complete configuration for my Traefik deployment and ingress.  Have you deployed Traefik with these yaml files?  The only thing I haven't included is the deployment for MetalLB.  I am unable to include the log files for the Traefik deployment since it's on an inaccessible network. 

With regards to the setup, can you suggest what I should do to simply my deployment?  I believe the deployment configuration I have is as simple as I can make it.  I have already tested removing the udp deployment but when I do that, the problem is fixed.  This of course is the issue I'm trying to solve:)

**Comment by shaggykp** [comment]
I'm not sure it was clear in my earlier comments, that the reason the traefik deployment was failing when there was more than 1 instance running is because there are 2 traefik services sharing the same external loadbalancer IP.  As I stated in my earlier comments, with the configuration I have provided, MetalLB should be able to support this.  I have also followed what the traefik documentation says about specifying UDP and TCP endpoints.
